bundles/org.openhab.binding.sonos/src/main/java/org/openhab/binding/sonos/internal/discovery/ZonePlayerDiscoveryParticipant.java

```diff
@@ -100,22 +100,26 @@ public Set<ThingTypeUID> getSupportedThingTypeUIDs() {
                         ignored = true;
                         break;
                     default:
+                        modelName = modelName.replaceAll("[^a-zA-Z0-9_]", "_");
```

Comment:
This seems inconsistent with the current hardcoded translations where spaces are simply stripped, for example "Arc SL" -> "ArcSL" while this would produce "Arc_SL" instead. If this would be consistent, cases for "One SL" and "Arc SL" could be dropped.

Reply:
All those for which a space is expected (like "Arc SL", 'One SL" are handled in switch cases ; the default handles all other cases and space is then not expected.

One solution to avoid the special cases for "Arc SL" or "One SL" would be to rename thing type ArcSL into Arc_SL but this is a breaking change so I do not want to do it and I prefer to do it that way.

So for me, the new code in the default case is OK.

Reply:
Okay, let me try to explain. It seems that the "standard" translation to thing type is by removing spaces, for example "Arc SL" to "ArcSL". But now default is to replace them by underscores. If this would be harmonized you wouldn't need the two special cases for "One SL" and "Arc SL" as this would be correctly handled by the default case.

If a new "Arc XY" would be introduced, would you add another case for it being translated to "ArcXY" or would you leave it for the default case to translate it to "Arc_XY"? If you prefer underscores for future additions and "Arc SL" and "One SL" are only legacy translations, then current code is fine.

Reply:
Yes, exactly, I don't want to break existing setup by changing thing types.
In the future, either I will add a new specific case in the discovery code or I will use the _ in the thing type. Adding one _ in future thing types would be the best option I believe.

Reply:
Ok, I understand what you would like. You would like that I replace non standard characters by "" rather than by "_".
I think you are right, iut is the best option.

Reply:
Ok I think it is now how you expect.

Reply:
Exactly.